assets/extensions/extension-actions.js

```diff
@@ -42,21 +45,37 @@ const ExtensionActions = ( { actions } ) => (
 export default ExtensionActions;
 
 /**
- * Get extension actions array.
+ * Extension actions hook.
  *
  * @param {Object} extension Extension object.
  *
  * @return {Array|null} Array of actions, or null if it's not a valid extension.
  */
-export const getExtensionActions = ( extension ) => {
+export const useExtensionActions = ( extension ) => {
+	const { updateExtensions } = useDispatch( EXTENSIONS_STORE );
+
 	if ( ! extension.product_slug ) {
 		return null;
 	}
 
 	let buttonLabel = '';
+	let buttonAction = () => {};
 
-	if ( extension.has_update ) {
+	if ( isLoadingStatus( extension.status ) ) {
+		buttonLabel = (
+			<>
+				<UpdateIcon
+					width="20"
+					height="20"
+					className="sensei-extensions__rotating-icon sensei-extensions__extension-actions__button-icon"
+				/>
+				{ __( 'Updating…', 'sensei-lms' ) }
```

Comment:
As already mentioned by @gkaragia , this part needs some refactor when implementing the install too. Maybe using generic stuff, or customizing the different options.